bin/optimizer.php

```diff
@@ -10,32 +10,20 @@
  * file that was distributed with this source code.
  */
 
-use Joli\GifExceptionBundle\Tests\app\src\Kernel;
-use Symfony\Bundle\FrameworkBundle\Console\Application;
-use Symfony\Component\Console\Input\ArgvInput;
-
-// if you don't want to setup permissions the proper way, just uncomment the following PHP line
-// read http://symfony.com/doc/current/book/installation.html#configuration-and-setup for more information
-// umask(0000);
-set_time_limit(0);
-
-/**
- * @var Composer\Autoload\ClassLoader
- */
-$loader = require __DIR__ . '/../vendor/autoload.php';
-
-$args = $_SERVER['argv'];
-
-// Strip application name
-array_shift($args);
-
-// Prepend command name
-array_unshift($args, 'jolicode:gifexception:optimize');
-
-// Prepend application name (ArgvInput strips it again so needs to be here)
-array_unshift($args, __DIR__ . '/optimizer.php');
-
-$input = new ArgvInput($args);
-$kernel = new Kernel('dev', false);
-$application = new Application($kernel);
-$application->run($input);
+use Joli\GifExceptionBundle\Command\GifOptimizerCommand;
+use Symfony\Component\Console\Application;
+
+if (file_exists(__DIR__ . '/../vendor/autoload.php')) {
+    $loader = require __DIR__ . '/../vendor/autoload.php';
+} elseif (file_exists(__DIR__ . '/../../../../vendor/autoload.php')) {
+    $loader = require __DIR__ . '/../../../../vendor/autoload.php';
+} else {
+    throw new \RuntimeException('Unable to load autoloader.');
+}
+
+$application = new Application('gifexception');
+$application->add(new GifOptimizerCommand());
+$application
+    ->setDefaultCommand('gifexception:optimize', true)
```

Comment:
I would use `SingleCommandApplication`, it's simpler

Reply:
That means to drop support for running the command on user's apps. I'm fine with it but I have no time to do it right now. Let's merge this PR for now. :+1: